### PR TITLE
misc(backend): Add call timeout to the backend as a parameter

### PR DESCRIPTION
**Pull request by Haskaris**
See #5087
I've changed to timeout parameter, before hard coded in frontend, now backend parameter in configfile (development.ini)

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/development.ini.sample [hunk 1]
```diff
@@ -870,7 +870,12 @@ caldav.radicale.headers.Access-Control-Expose-Headers = Etag
 ## Jitsi Meet configuration parameters
 ; call.jitsi_meet.url =
 # example:
-; call.jitsi_meet.url = https://meet.jit.si
+# call.jitsi_meet.url = https://meet.jit.si
+
+## Time (in ms) before the call is considered unanswered
+; call.unanswered_timeout =
```
Comment (PhilippeAccorsi) [inline backend/development.ini.sample:876]: Tracim not have default value ?

If default value exist, good syntax is `; call.unanswered_timeout = 30000`

File: backend/development.ini.sample [hunk 2]
```diff
@@ -870,7 +870,12 @@ caldav.radicale.headers.Access-Control-Expose-Headers = Etag
...
+# example:
+# call.unanswered_timeout = 30000
```
Comment (PhilippeAccorsi) [inline backend/development.ini.sample:878]: Its better to give an example with not default value. 
~~~
# Example: timeout after 60s
# call.unanswered_timeout = 60000
~~~

File: backend/doc/setting.md [hunk 3]
```diff
@@ -473,8 +473,8 @@ To enable online edition on Tracim and allow communication with your edition sof
 
 First you need to enable the edition on the API:
 
-    collaborative_document_edition.activated = True
-    collaborative_document_edition.software = collabora`
+    app_enabled = collaborative_document_edition
```
Comment (PhilippeAccorsi) [inline backend/doc/setting.md:476]: Do you sure about this syntax ?

File: backend/tests_configs.ini [hunk 4]
```diff
@@ -921,6 +921,15 @@ collaborative_document_edition.software = collabora
...
 collaborative_document_edition.collabora.base_url = http://localhost:9980
 
+[call_jitsi_test]
+app.enabled = contents/thread,contents/file,contents/html-document,contents/folder,upload_permission,share_content
+website.base_url = http://localhost:6543
+auth_types = internal
+call.enabled = True
+call.provider = jitsi_meet
+call.jitsi_meet.url = https://meet.jit.si
+call.unanswered_timeout = 20000
```
Comment (PhilippeAccorsi) [inline backend/tests_configs.ini:931]: why this config not use default value ?

**Review by PhilippeAccorsi — CHANGES_REQUESTED** [review]

**Review by raphj — APPROVED** [review]

File: backend/doc/setting.md [hunk 5]
```diff
@@ -496,7 +496,34 @@ But you can change the default directory to use your templates files:
...
+
+### Configuring Tracim in `development.ini`
+To enable the call feature on Tracim.
+
+First you need to enable the call feature:
+
+    call.enabled = True
+
+#### Mandatory parameters
+Then select a profider:
```
Comment (grignards) [inline backend/doc/setting.md:511]:
```suggestion
Then select a provider:
```

File: backend/doc/setting.md [hunk 6]
```diff
@@ -496,7 +496,34 @@ But you can change the default directory to use your templates files:
...
+Tracim is only supporting jitsi so far.
+
+Once you have selected a provider you will have to select an url:
+
+    call.jitsi_meet.url = https://meet.jit.si
+
+#### Facultative parameters
+You can specify the time (in ms) before the call is considered unanswered:
+
+    call.unanswered_timeout = 30000
```
Comment (grignards) [inline backend/doc/setting.md:524]: It is a bit strange to ask for milliseconds here, I think most time configuration parameters are given in seconds.

**Review by gbonaspetti — APPROVED** [review]
ok for frontend part

File: backend/doc/setting.md (hunk 3 above)
Comment (inkhey) [reply backend/doc/setting.md:476]: The syntax is incorrect:
- user should **add** an app to existing one, not set only collaborative_document_edition
- **collaborative_document_edition** app need **contents/file** app to work (it's a specific case, most apps are not depend from others)

File: backend/development.ini.sample (hunk 1 above)
Comment (m2gi-perrier5) [reply backend/development.ini.sample:876]: Indeed, Tracim got a default value of 30 seconds. I'll change the syntax.

Comment (Haskaris) [reply backend/development.ini.sample:876]: Indeed, Tracim got a default value of 30 seconds. I'll change the syntax.

File: backend/development.ini.sample (hunk 2 above)
Comment (Haskaris) [reply backend/development.ini.sample:878]: Ok, I'll change the example.

File: backend/doc/setting.md (hunk 3 above)
Comment (Haskaris) [reply backend/doc/setting.md:476]: Indeed, I was aware we have to add the app, I thought about ``app_enabled += collaborative_document_edition``.
I wasn't aware of the second point.
I can change for another syntax after a discussion, so we can agree on a solution. Or I can revert this change, since it was a quick fix (I spotted the mistake) and create a new issue to update ``setting.md``.

File: backend/doc/setting.md (hunk 6 above)
Comment (Haskaris) [reply backend/doc/setting.md:524]: Indeed, I thought milliseconds were more flexible, but there are no configuration parameters in milliseconds. I'll change it to be in seconds (example too).

File: backend/tests_configs.ini (hunk 4 above)
Comment (Haskaris) [reply backend/tests_configs.ini:931]: It's for test purpose, in this test we verify that we got the `call.unanswered_timeout` parameter with the value specified in test_configs.ini
In my opinion it is be better to verify another value than the default one since this is not a mandatory parameter. If the `call.unanswered_timeout` parameter is not present into the config file (development.ini) we just use the default value

File: backend/doc/setting.md [hunk 7]
```diff
@@ -471,10 +471,25 @@ With a Collabora host, `<collabora_host>` may look like `collaboradomain.ndd` or
...
+
+    Before:
+    app.enabled = contents/thread,contents/html-document,contents/folder,agenda,collaborative_document_edition
+
+    After:
+    app.enabled = contents/thread,contents/file,contents/html-document,contents/folder,agenda,collaborative_document_edition
+
+Once the application enabled, you will have to specify which software you desire to use. So far Tracim only support Collabora.
+
+    collaborative_document_edition.software = collabora
```
Comment (PhilippeAccorsi) [inline backend/doc/setting.md:492]: to make sure : this is not directly related to the call feature, right.

File: backend/tests_configs.ini [hunk 8]
```diff
@@ -921,6 +921,15 @@ collaborative_document_edition.software = collabora
...
 collaborative_document_edition.collabora.base_url = http://localhost:9980
 
+[call_jitsi_test]
+app.enabled = contents/thread,contents/file,contents/html-document,contents/folder,upload_permission,share_content
+website.base_url = http://localhost:6543
+auth_types = internal
+call.enabled = True
+call.provider = jitsi_meet
+call.jitsi_meet.url = https://meet.jit.si
+call.unanswered_timeout = 20
```
Comment (PhilippeAccorsi) [inline backend/tests_configs.ini:931]: Is it not better to use same timeout value like we have in code ?

File: backend/tracim_backend/tests/functional/test_system.py [hunk 9]
```diff
@@ -279,6 +279,20 @@ def test_api__get_config__ok_200__nominal_case(self, patched_get, web_testapp):
...
+class TestConfigCallJitsi(object):
+    """
+    Tests for /api/system/config
+    """
+
+    def test_api__get_config__ok_200__nominal_case(self, web_testapp):
+        web_testapp.authorization = ("Basic", ("admin@admin.admin", "admin@admin.admin"))
+        res = web_testapp.get("/api/system/config", status=200)
+        assert res.json_body["call__enabled"] is True
+        assert res.json_body["call__unanswered_timeout"] == 20
```
Comment (PhilippeAccorsi) [inline backend/tracim_backend/tests/functional/test_system.py:293]: same question here about timeout

**Review by PhilippeAccorsi — CHANGES_REQUESTED** [review]

File: functionnal_tests/cypress/integration/workspace/create_workspace_spec.js [hunk 10]
```diff
@@ -21,7 +21,7 @@ describe('Workspace', () => {
       .get('.newSpace__button .btn').last().should('be.visible').click()
   }
 
-  it('should create a new space while no others are created', () => {
+  it.skip('should create a new space while no others are created', () => {
```
Comment (PhilippeAccorsi) [inline functionnal_tests/cypress/integration/workspace/create_workspace_spec.js:24]: Do you know if an issue exist about stabilization of this test ? (make sure other dev know this problem)

File: functionnal_tests/cypress/integration/workspace/create_workspace_spec.js [hunk 11]
```diff
@@ -53,7 +53,7 @@ describe('Workspace', () => {
       }
     })
 
-    it('should scroll to the new workspace in the sidebar', () => {
+    it.skip('should scroll to the new workspace in the sidebar', () => {
```
Comment (PhilippeAccorsi) [inline functionnal_tests/cypress/integration/workspace/create_workspace_spec.js:56]: same here

File: backend/doc/setting.md (hunk 7 above)
Comment (Haskaris) [reply backend/doc/setting.md:492]: Yes, it's a fix of an error that I just spotted. I didn't get an answer on [this comment](https://github.com/tracim/tracim/pull/5089#discussion_r749285244), but I had a discussion with @inkhey who had no real solution about "how explain how to add an application in the doc".

File: backend/tests_configs.ini (hunk 8 above)
Comment (Haskaris) [reply backend/tests_configs.ini:931]: In my opinion it is be better to verify another value than the default one since this is not a mandatory parameter. If the call.unanswered_timeout parameter is not present into the config file (development.ini) we just use the default value.

File: backend/tracim_backend/tests/functional/test_system.py (hunk 9 above)
Comment (Haskaris) [reply backend/tracim_backend/tests/functional/test_system.py:293]: The value have to be the same as the one in the file tests_configs.ini

File: functionnal_tests/cypress/integration/workspace/create_workspace_spec.js (hunk 10 above)
Comment (Haskaris) [reply functionnal_tests/cypress/integration/workspace/create_workspace_spec.js:24]: Yup #5091

File: functionnal_tests/cypress/integration/workspace/create_workspace_spec.js (hunk 11 above)
Comment (Haskaris) [reply functionnal_tests/cypress/integration/workspace/create_workspace_spec.js:56]: Same ticket

File: backend/doc/setting.md (hunk 7 above)
Comment (PhilippeAccorsi) [reply backend/doc/setting.md:492]: ok :)

File: backend/tracim_backend/tests/functional/test_system.py (hunk 9 above)
Comment (PhilippeAccorsi) [reply backend/tracim_backend/tests/functional/test_system.py:293]: ok so maybe you can decrease value to not slow the test (its an idea)

File: backend/tests_configs.ini (hunk 8 above)
Comment (PhilippeAccorsi) [reply backend/tests_configs.ini:931]: ok so maybe you can decrease value to not slow the test (its an idea)

File: backend/tracim_backend/tests/functional/test_system.py (hunk 9 above)
Comment (PhilippeAccorsi) [reply backend/tracim_backend/tests/functional/test_system.py:293]: ok

File: functionnal_tests/cypress/integration/workspace/create_workspace_spec.js (hunk 10 above)
Comment (PhilippeAccorsi) [reply functionnal_tests/cypress/integration/workspace/create_workspace_spec.js:24]: ok

File: functionnal_tests/cypress/integration/workspace/create_workspace_spec.js (hunk 11 above)
Comment (PhilippeAccorsi) [reply functionnal_tests/cypress/integration/workspace/create_workspace_spec.js:56]: ok

**Review by inkhey — APPROVED** [review]

File: backend/tests_configs.ini (hunk 8 above)
Comment (PhilippeAccorsi) [reply backend/tests_configs.ini:931]: After discuss with  Haskaris, I understand what is done in this test. So no problem here.

**Review by PhilippeAccorsi — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch
